packages/boot/README.md

```diff
@@ -0,0 +1,90 @@
+# @loopback/boot
+
+A collection of Booters for LoopBack Applications
+
+# Overview
+
+A Booter is a Class that can be bound to an Application and is called
+to perform a task before the Application is started. A Booter may have multiple
+phases to complete its task.
+
+An example task of a Booter may be to discover and bind all artifacts of a
+given type.
+
+A BootStrapper is needed to manage the Booters and to run them. This is packaged
+in BootComponent. Add `BootComponent` to your `Application` to use `Booters`.
+
+## Installation
+
+```shell
+$ npm i @loopback/boot
+```
+
+## Basic Use
+
+```ts
+import {Application} from '@loopback/core';
+import {ControllerBooter, RepositoryBooter, BootComponent} from '@loopback/boot';
+const app = new Application();
+app.component(BootComponent);
+app.booter(ControllerBooter); // register booter
+await app.boot({
```

Comment:
I am still standing behind my earlier comment that `boot` should not be part of the built-in `Application` class, I am not sure why it has been collapsed by GitHub.

IMO, this example should be showing how to use `BootStrapper` in isolation.

Reply:
We have a few built-in helper methods on `Application` for some of the core artifacts, such as: `app.controller`, and `app.component`. They are simply wrappers to call `context.bind()` by conventions. 

@bajtos Do you think boot/booter is secondary?

Reply:
> Do you think boot/booter is secondary?

I am not sure, to be honest. Ideally, we don't want `core` to depend on `boot`, similarly how `core` does not depend on `rest` nor `repository`.

At the same time, it may make sense to make it easy for 3rd-party components to register booters via `app.booter()` API. Unless we find a way how to allow 3rd-party components to register booters with the BootStrapper instance that will be used by `app.boot()`.

Thoughts?

As for `app.boot()`, I am starting to see a value of having this API contract defined right in core (but not implemented via loopback boot!) - see https://github.com/strongloop/loopback-next/pull/858#issuecomment-360719908

> BTW, with boot in place, we'll see less explicit `app.controller`/`app.component`/`app.*` usage.

Agreed. However, these helpers should be still available for our tests, for people who want to wire up their applications their own way, and they should be used by booters too.